app/rpc/manager.go

```diff
@@ -196,7 +196,7 @@ func (m *Manager) notifyVirtualSelectedParentChainChanged(virtualChangeSet *exte
 	defer onEnd()
 
 	notification, err := m.context.ConvertVirtualSelectedParentChainChangesToChainChangedNotificationMessage(
-		virtualChangeSet.VirtualSelectedParentChainChanges)
+		virtualChangeSet.VirtualSelectedParentChainChanges, true)
```

Comment:
You're populating the notification with tx data even if the user didn't subscribe to RPC notifications. I guess it could be a heavy operation.